crates/garble-core/src/lib.rs

```diff
@@ -37,6 +43,22 @@ const MAX_BATCH_SIZE: usize = 4 * KB;
 /// This puts an upper limit on that waste.
 pub(crate) const DEFAULT_BATCH_SIZE: usize = MAX_BATCH_SIZE / BYTES_PER_GATE;
 
+/// The initial gate id.
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
This can be `Copy`, then we do not need to clone it.
```suggestion
#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
```